tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/runtime/api/RuntimeAPITest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://wso2.com).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ballerinalang.test.runtime.api;
+
+import org.ballerinalang.test.BaseTest;
+import org.ballerinalang.test.context.BMainInstance;
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.ballerinalang.test.util.AssertionUtils;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.ballerina.projects.util.ProjectConstants.USER_DIR;
+import static io.ballerina.runtime.api.constants.RuntimeConstants.BALLERINA_HOME;
+import static org.ballerinalang.test.util.SQLDBUtils.deleteDirectory;
+
+/**
+ * Test class to test the functionality of Ballerina runtime APIs for invoking functions.
+ *
+ * @since 2201.9.0
+ */
+public class RuntimeAPITest extends BaseTest {
+
+    private static final String testFileLocation = Paths.get("src", "test", "resources", "runtime.api")
+            .toAbsolutePath().toString();
+    private static final Path javaSrcLocation = Paths.get("src", "test", "java", "org", "ballerinalang",
+            "test", "runtime", "api").toAbsolutePath();
+    private BMainInstance bMainInstance;
+
+    @BeforeClass
+    public void setup() throws BallerinaTestException {
+        bMainInstance = new BMainInstance(balServer);
+        Path sourceRoot = Paths.get(testFileLocation, "function_invocation").toAbsolutePath();
+        Map<String, String> envProperties = new HashMap<>();
+        bMainInstance.addJavaAgents(envProperties);
+        bMainInstance.runMain("build", new String[0], envProperties, new String[0], null,
+                sourceRoot.toString());
+    }
+
+    @Test
+    public void testRuntimeAPIsForBalFunctionInvocation() throws BallerinaTestException {
+        Path jarPath = Paths.get(testFileLocation, "function_invocation", "target", "bin",
```

Comment:
Shall we refactor this method into multiple private method calls for readability?
Something like `compileJavaSource()`, `unzipJar()`, `packTestJar()` and `executeAndValidateTestJar()`.